### PR TITLE
Add publish rate parameter

### PR DESCRIPTION
**Pull request by areyna1**
added parameter for changing publish rate

### CODE REVIEW

File: src/bennu/executables/bennu-simulink-provider/main.cpp [hunk 1]
```diff
@@ -121,11 +121,13 @@ struct Dto
 class BennuSimulinkProvider : public Provider
 {
 public:
-    BennuSimulinkProvider(const Endpoint& serverEndpoint, const Endpoint& publishEndpoint, bool debug) :
+    double publishRate_setting;
```
Comment (GhostofGoes) [inline src/bennu/executables/bennu-simulink-provider/main.cpp:124]: Add a newline after the variable and before the constructor

File: src/bennu/executables/bennu-simulink-provider/main.cpp [hunk 2]
```diff
@@ -284,7 +286,8 @@ class BennuSimulinkProvider : public Provider
         {
             publishData();
             std::cout << std::flush;
-            std::this_thread::sleep_for(std::chrono::seconds(1));
+            // std::this_thread::sleep_for(std::chrono::seconds(1));
```
Comment (GhostofGoes) [inline src/bennu/executables/bennu-simulink-provider/main.cpp:289]: Remove commented out code

File: src/bennu/executables/bennu-simulink-provider/main.cpp [hunk 3]
```diff
@@ -284,7 +286,8 @@ class BennuSimulinkProvider : public Provider
...
+            std::this_thread::sleep_for(durationToDuration(this->publishRate_setting));
```
Comment (GhostofGoes) [inline src/bennu/executables/bennu-simulink-provider/main.cpp:290]: where is durationToDuration defined? Did you forget to include in the PR?

File: src/bennu/executables/bennu-simulink-provider/main.cpp [hunk 4]
```diff
@@ -333,6 +336,7 @@ int main(int argc, char** argv)
         ("debug", po::bool_switch(&debug), "print debugging information")
         ("server-endpoint", po::value<std::string>()->default_value("tcp://127.0.0.1:5555"), "server listening endpoint")
         ("publish-endpoint", po::value<std::string>()->default_value("udp://239.0.0.1:40000"), "publishing endpoint");
+        ("publish-rate", po::value<double>()->default_value(0.1), "rate at which updates published from provider to simulation");
```
Comment (GhostofGoes) [inline src/bennu/executables/bennu-simulink-provider/main.cpp:339]: grammar, change to `rate at which updates are published from the provider to the simulation`

File: src/bennu/executables/bennu-simulink-provider/main.cpp [hunk 5]
```diff
@@ -348,6 +352,8 @@ int main(int argc, char** argv)
     sEndpoint.str = vm["server-endpoint"].as<std::string>();
     pEndpoint.str = vm["publish-endpoint"].as<std::string>();
     BennuSimulinkProvider bsp(sEndpoint, pEndpoint, debug);
```
Comment (GhostofGoes) [inline src/bennu/executables/bennu-simulink-provider/main.cpp:354]: bsp is being initialized twice, remove the old initialization without publishRate

**Review by GhostofGoes — CHANGES_REQUESTED** [review]
Please rename your PR to be the feature you're adding.

File: src/bennu/executables/bennu-simulink-provider/main.cpp (hunk 3 above)
Comment (areyna1) [reply src/bennu/executables/bennu-simulink-provider/main.cpp:290]: Included function declaration above BennuSimulinkProvider class

**Review by GhostofGoes — APPROVED** [review]
Looks good. One thought for future, make `publishRate_setting` type of auto, and call `durationToDuration` in the constructor. Calling it every time the thread sleeps is unnecessary since the value never changes.

**Comment by GhostofGoes** [comment]
@mgaliar when you have time, can you give this a cursury glance and blessing before I merge it?

**Review by mgaliar — APPROVED** [review]

**Comment by mgaliar** [comment]
Looks good!

**Comment by GhostofGoes** [comment]
One note: this changes the default behavior from polling every second (if I read the code correctly) to polling every 100ms. Shouldn't impact anything negatively.

**Pull request merged**